src/main/java/com/fastmail/jstpierre/petmanager/predicates/Predicates.java

```diff
@@ -0,0 +1,43 @@
+package com.fastmail.jstpierre.petmanager.predicates;
+
+import com.fastmail.jstpierre.petmanager.entities.petowners.PetOwner;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.function.BiPredicate;
+
+public class Predicates {
+    public static final BiPredicate<PetOwner, String> ownerHasPetByPetId = (owner, petId) ->
+            owner.getPets()
+                    .stream()
+                    .anyMatch(pet -> pet.getId().equals(petId));
+
+    public static final BiPredicate<PetOwner, String> ownerHasPetByPetName = (owner, petName) ->
+            owner.getPets()
+                    .stream()
+                    .anyMatch(pet -> pet.getName().equals(petName));
+
+    public static final BiPredicate<PetOwner, Integer> ownerHasPetByPetAge = (owner, petAge) ->
+            owner.getPets()
+                    .stream()
+                    .anyMatch(pet -> pet.getAge().equals(petAge));
+
+    public static final BiPredicate<PetOwner, String> ownerHasPetByPetToyName = (owner, petToyName) ->
+            owner.getPetToys()
+                    .stream()
+                    .anyMatch(petToy -> petToy.getName().equals(petToyName));
+
+    public static final BiPredicate<List<PetOwner>, PetOwner> ownersLivesAtAddress = (owners, owner) ->
+            owner.getStreetAddress().equals(getOwnerByAddress(
+                    owners, owner.getStreetAddress()
+            ));
+
+    private static String getOwnerByAddress(List<PetOwner> petOwners, String address) {
+        Optional<String> petOwnerAddress = petOwners.stream()
+                .map(PetOwner::getStreetAddress)
+                .filter(streetAddress -> streetAddress.equals(address))
+                .findFirst();
+
+        return petOwnerAddress.orElse(null);
```

Comment:
Peut être remplacé par une oneliner :
```suggestion
        return petOwners.stream()
                .map(PetOwner::getStreetAddress)
                .filter(streetAddress -> streetAddress.equals(address))
                .findFirst()
                .orElse(null);
```